backend/modules/pdf_processing/router.py

```
@@ -0,0 +1,287 @@
+                "role": "Marketing Lead",
+                "background": "Marketing professional focused on brand strategy and customer engagement.",
+                "primary_goals": ["Build brand awareness", "Drive customer acquisition", "Develop marketing strategies"],
+                "personality_traits": {"analytical": 6, "creative": 9, "assertive": 7, "collaborative": 8, "detail_oriented": 6}
+            }
+        ]
+    }
+
+
+@router.post("/parse-pdf-with-progress")
+async def parse_pdf_with_progress_route(
+    file: UploadFile = File(...),
```

Ruff (0.14.5)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)